### PR TITLE
EZP-30626 Remove Nameable field type interface

### PR DESCRIPTION
**Pull request by MagdalenaZuba**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://jira.ez.no/browse/EZP-30626
| Versions      | master

Remove Nameable field type interface in the main Field Type documentation.
Creating a Tweet Field Type tutorial will follow in a separate PR.

### CODE REVIEW

File: docs/tutorials/field_type/3_implement_the_tweet_type_class.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ A Type class must implement `eZ\Publish\SPI\FieldType\FieldType` ("Field Type in
 
 All native Field Types also extend the `eZ\Publish\Core\FieldType\FieldType` abstract class that implements this interface and provides implementation facilities through a set of abstract methods of its own. In this case, Type classes implement a mix of methods from the Field Type interface and from the abstract Field Type.
 
-The recommended way to allow the Field Type to be able to generate content name is through implementing the `eZ\Publish\SPI\FieldType\Nameable` interface.
+The recommended way to allow the Field Type to be able to generate content name is through the `FieldType::getName()`.
```
Comment (adamwojs) [inline docs/tutorials/field_type/3_implement_the_tweet_type_class.md]: In 3.0 there is no other way then `eZ\Publish\Core\FieldType\FieldType::getName`

File: docs/api/field_type_type_and_value.md [hunk 2]
```diff
@@ -45,18 +45,14 @@ This will also apply to all user interfaces and the REST API, which therefore mu
...
-The `eZ\Publish\SPI\FieldType\Nameable::getFieldName` method will be used to retrieve the name.
+To be able to retrieve a Content item name when the Field is part of a name schema or a URL schema,
+use `eZ\Publish\Core\FieldType\FieldType::getName` method.
 
 ``` yaml
 ezpublish.fieldType.ezobjectrelation.nameable_field:
     class: '%ezpublish.fieldType.ezobjectrelation.nameable_field.class%'
     arguments:
       - '@ezpublish.spi.persistence.cache.contentHandler'
-    tags:
```
Comment (adamwojs) [inline docs/api/field_type_type_and_value.md]: The whole code snippet should be deleted.

File: docs/api/field_type_type_and_value.md [hunk 3]
```diff
@@ -45,19 +45,8 @@ This will also apply to all user interfaces and the REST API, which therefore mu
...
-``` yaml
-ezpublish.fieldType.ezobjectrelation.nameable_field:
-    class: '%ezpublish.fieldType.ezobjectrelation.nameable_field.class%'
-    arguments:
-      - '@ezpublish.spi.persistence.cache.contentHandler'
-    tags:
-        - {name: ezpublish.fieldType.nameable, alias: ezobjectrelation}
-```
+To be able to retrieve a Content item name when the Field is part of a name schema or a URL schema,
+use `eZ\Publish\Core\FieldType\FieldType::getName` method.
```
Comment (alongosz) [inline docs/api/field_type_type_and_value.md]: We need to hint against SPI interface here: `eZ\Publish\SPI\FieldType\FieldType::getName`. Moreover this is no longer optional, so I would rephrase this sentence, something like
```
Field name is used when the Field is part of a name schema or a URL schema to generate Content item name or URL alias respectively. It is retrieved by calling `eZ\Publish\SPI\FieldType\FieldType::getName` method which must be implemented.
```
feel free of course to fix that the way it suits you ;)

File: docs/tutorials/field_type/3_implement_the_tweet_type_class.md [hunk 4]
```diff
@@ -11,7 +11,7 @@ A Type class must implement `eZ\Publish\SPI\FieldType\FieldType` ("Field Type in
 
 All native Field Types also extend the `eZ\Publish\Core\FieldType\FieldType` abstract class that implements this interface and provides implementation facilities through a set of abstract methods of its own. In this case, Type classes implement a mix of methods from the Field Type interface and from the abstract Field Type.
 
-The recommended way to allow the Field Type to be able to generate content name is through implementing the `eZ\Publish\SPI\FieldType\Nameable` interface.
+To allow the Field Type to retrieve content name use the `\eZ\Publish\Core\FieldType\FieldType::getName` method.
```
Comment (alongosz) [inline docs/tutorials/field_type/3_implement_the_tweet_type_class.md]:
```suggestion
To allow the Field Type to retrieve content name implement the `eZ\Publish\SPI\FieldType\FieldType::getName` method.
```
Please also decide it you want to precede all FQCNs with `\` or not. Sometimes it is preceded sometimes it is not. For the documentation it's just a matter of convention for readability.

Comment (MagdalenaZuba) [reply docs/tutorials/field_type/3_implement_the_tweet_type_class.md]: This change  was for consistency purpose exactly :)
